### ISSUE TITLE
JPlayer Picture in Picture playback on Android

### ISSUE THREAD

**Issue by GreenDragonTavern** [WWBN/AVideo#7897]
Not sure what licensing is involved with this feature request and hopefully mobile friendliness is a priority with AVideo...so with that said. Not sure about iOS but there's documented support for Android. AVideo would be no different than YouTube on Android phones which allows video PiP while using other apps on phone. The cats' meow would be once AVideo is backgrounded while playing a video it invokes PiP automatically just like YouTube.

**Comment by DanielnetoDotCom** [comment]
Awesome idea, try this update

**Comment by GreenDragonTavern** [comment]
Cool! Had to turn phone landscape to get PiP button to show but worked like a champ. What would it take for AVideo to do this automatically depending on focus, ie. automatically invoke PiP when AVideo loses focus?

**Comment by DanielnetoDotCom** [comment]
It suppose to go to pip automatically if the video is playing and loose focus

**Comment by GreenDragonTavern** [comment]
Negative, tried Edge and Chrome browsers on my Samsung phone and it stopped playing video as soon as it lost focus.

**Comment by GreenDragonTavern** [comment]
On Windows 11, my laptop, it did invoke PiP upon losing focus. There is an issue though...if I jump back to AVideo tab it disabled PiP and video was back to normal embed, which is perfect but when I selected another open tab it did not invoke PiP a second time.

**Comment by DanielnetoDotCom** [comment]
on mobile you must go fullscreen to get the pip when you loose focus

**Comment by DanielnetoDotCom** [comment]
Uncaught (in promise) DOMException: Failed to execute 'requestPictureInPicture' on 'HTMLVideoElement': Must be handling a user gesture if there isn't already an element in Picture-in-Picture.
    at e.requestPictureInPicture (https://tutorialsavideocom.cdn.ypt.me/node_modules/video.js/dist/video.min.js?cache=1676423657_1676423657:20:130207)
    at e.techGet_ (https://tutorialsavideocom.cdn.ypt.me/node_modules/video.js/dist/video.min.js?cache=1676423657_1676423657:20:159254)
    at e.requestPictureInPicture (https://tutorialsavideocom.cdn.ypt.me/node_modules/video.js/dist/video.min.js?cache=1676423657_1676423657:20:165608)
    at HTMLDocument.<anonymous> (https://tutorials.avideo.com/video/78/welcome-to-avideo-platform:4223:9)
e.requestPictureInPicture @ video.min.js?cache=1676423657_1676423657:20
e.techGet_ @ video.min.js?cache=1676423657_1676423657:20
e.requestPictureInPicture @ video.min.js?cache=1676423657_1676423657:20
(anonymous) @ welcome-to-avideo-platform:4223
welcome-to-avideo-platform:4219 Picture-in-Picture visibilitychange visible

**Comment by GreenDragonTavern** [comment]
Is there a specific reason full screen is necessary on mobile? Landscape is cumbersome on mobile since portrait is the norm and jumping around on phone for the most part involves notifications and addressing them. I listen to YouTube all the time on phone and it's a notification I'm responding to that makes their app so awesome. Get an email, read it and never miss a beat on what's playing on YouTube at the time. I'd personally create custom playlists in AVideo if it did the same and never use YouTube again except to upload to AVideo 😁

**Comment by DanielnetoDotCom** [comment]
> Is there a specific reason full screen is necessary on mobile? 

not really, I guess this is a browser limitation.
youtube is different, they treat themself differently.

**Comment by GreenDragonTavern** [comment]
My guess would be the difference between Web JPlayer and Android/iOS JPlayer. They're probably different in not only how they're displayed on device but how JPlayer API interacts or doesn't.

**Comment by DanielnetoDotCom** [comment]
do you mean jplayer can go to the pip in mobile without fullscreen?

try to pause and unpause the video before lose focus

**Comment by GreenDragonTavern** [comment]
Nope, fullscreen on phone is only way to invoke PiP. The most functional method I've found on phone so far is to go fullscreen, manually press PiP and then shift phone back to portrait. Otherwise letting PiP invoke itself is spotty, at least on my Samsung.
